addons/web/static/src/views/list/list_renderer.scss

```
@@ -340,7 +340,9 @@
                 }
                 &.o_field_monetary input {
                     width: 0; // override width: 100px and let flex makes this input grow
-                    margin: 0 4px;
+                }
+                &.o_field_monetary .o_input.d-flex {
```

```suggestion
                &.o_field_monetary span.o_input {
```
or add a specific classname on that span, and use it here instead of o_input

---

also I'm not sure about using class o_input on the span, which rules do you require from it?

---

it's useful for some positioning rules